### PR TITLE
Update CONTRIBUTING.md -- Java

### PR DESCRIPTION
**Pull request by pdowler**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,9 +21,44 @@ In OpenCADC, the publicly released code lives in the main (or master) branch of
 
 ### Java Development Guidelines
 
-TODO
-* checkstyle
-* CI
+Java code is licenced under the Affero GPL version 3 (AGPL-3.0) and each file originally created by CADC staff includes a standard bilingual copyright notice with `(C) Government of Canada ...`. Files originally created and contributed should be under the same license and may/should/usually contain their own suitable `(C)` claim; copyright is assumed to be assigned (to GoC) if the standard bilingual notice if included. **TBD**
+
```
Comment (brianmajor) [inline CONTRIBUTING.md:25]: Perhaps this licence info applies to all opencadc software and could be promoted out of this section?
There is also the LICENCE file that could be referenced here.
I don't think we need to be overly specific.
Should we remove the TBD?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,9 +21,44 @@ In OpenCADC, the publicly released code lives in the main (or master) branch of
...
+#### Code Style
+Java coding style is specified and validated using `checkstyle` via the <a href="https://github.com/opencadc/core/tree/main/cadc-quality">cadc-quality</a> library. Each repository includes this library in the test classpath and configures checkstyle via the repository `opencadc.gradle` file. Developers can run checkstyle locally using `gradle checkstyleMain` (we currently do not enforce checkstyle on test code **TBD**) to make sure they comply and **should** use the checkstyle configuration from `cadc-quality` to configure their development environment (IDE) where possible. 
+
+This is currently checkstyle-8.2 -- pretty old -- and has some defiencies that need to be addressed. In addition, `gradle javadoc` must have no errors and ideally no warnings (technical debt: most older code has lots of warnings). 
+
+The current code style checking really only covers small scale syntax and not larger scale code organisation and design. General principles:
+* clear code so comments are not needed or can be minimised
+* judicious comments to capture tricky details, especially intentionally ignored some detail
```
Comment (brianmajor) [inline CONTRIBUTING.md]: I'm not understanding the phrase `especially intentionally ignored some detail`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,9 +21,44 @@ In OpenCADC, the publicly released code lives in the main (or master) branch of
...
+* judicious comments to clarify larger scale logic in complex methods
+* constructor args for required values; final for immutable fields, prefer public fields when null-able
+* polymorphism: only when necessary
+* TODO: `ca.nrc.cadc` vs `org.opencadc` java packages
+* TODO: more general principles
+* TODO: improve checkstyle to avoid IDE-code-format-wars
```
Comment (brianmajor) [inline CONTRIBUTING.md:50]: I don't care a whole lot, but the TODOs could go in as issues in this repo rather than be in this doc.

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (pdowler) [reply CONTRIBUTING.md:25]: Could be promoted but needs to be somewhere. I don't know if the convention to put this in every file is the same in python, notebooks, or other kinds of "source" so only wanted to address it in java.

Also, we do need some stance on copyright, which I was saying is "assigned to GoC" if it has  the std header and if someone contributes a new file they can retain (C).... multiple copyright holders is always a bit of a pain and we don't want multiple at the individual file level. So a bug fix always assigns copyright and the other extreme would be a new module in the repo (eg tap.git/cadc-tap-server-bigquery) where it's easy enough to have a completely different copyright holder (I would expect it).
New files in tap.git/cadc-tap-server? I'd prefer copyright assignment to GoC but will need a real life case to figure it out.

File: CONTRIBUTING.md (hunk 2 above)
Comment (pdowler) [reply CONTRIBUTING.md]: bad grammar, should have been `intentionally ignoring some detail`.... I meant that if some code is simple and the author knows they are ignoring some detail that will cause it to fail that should be noted, but maybe this is too deep in the weeds for this document. Will simplify.

File: CONTRIBUTING.md (hunk 1 above)
Comment (pdowler) [reply CONTRIBUTING.md:25]: clarified but retained for now

**Review by brianmajor — APPROVED** [review]

**Pull request merged**